### PR TITLE
Restructure the document as RFC9093-bis

### LINKED ISSUE #32: Add more line-coding identities
See https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/issues/100

### LINKED ISSUE #34: Relationship with RFC9093
Need to clarify whether this draft updates or obsoletes RFC9093

Obsoleting RFC9093 will have an impact on the I-D text and YANG model

See discussion during IETF 112: https://datatracker.ietf.org/meeting/112/materials/minutes-112-ccamp-00.txt

### PR DESCRIPTION
**Pull request by italobusi**
Fix #34 : Updated the document to become an RFC9093-bit

Fix #32  :
- added bitrate to the common-explicit-mode grouping

Support resolution of https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/issues/99:
- added the following typedef
  - snr-or-null
  - decimal-2-digits
  - decimal-2-digits-or-null
  - power-in-db
  - power-in-db-or-null
  - power-in-dbm
  - power-in-dbm-or-null
- Updated types with union with type empty when applicable

Fix https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/issues/35
- added grouping l0-path-constraints
- added grouping l0-path-properties

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>
Co-Authored-By: Dieter Beller <Dieter.Beller@nokia.com>

### CODE REVIEW

File: ietf-layer0-types-ext.yang [hunk 1]
```diff
@@ -613,17 +687,17 @@ module ietf-layer0-types-ext {
...
+      type power-in-db-or-null;
       config false;
       mandatory true;
       description
         "Maximum acceptable accumulate polarization dependent loss";
     }
     leaf penalty {
-      type uint8;
+      type union {
+        type uint8;
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:698]: Shouldn't penalty be a decimal value ?

File: ietf-layer0-types-ext.yang [hunk 2]
```diff
@@ -648,10 +722,18 @@ module ietf-layer0-types-ext {
...
       }
       config false;
-      description "Bit rate/line coding of optical tributary signal";
+      description
+        "Bit rate/line coding of the optical tributary signal.";
       reference 
         "ITU-T G.698.2 section 7.1.2";
     }
+    leaf bitrate {
+      type uint16;
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:731]: shouldn't this be a decimal value ?
or maybe this value refers to the gross bitrate: eg 100, 200, ....

File: ietf-layer0-types-ext.yang [hunk 3]
```diff
@@ -684,10 +766,10 @@ module ietf-layer0-types-ext {
       uses cd-pmd-penalty ;
     }
     leaf max-diff-group-delay  {
-           type int32;
-           config false;
-           description "Maximum Differential group delay of this mode
-                        for this lane";
+      type int32;
```
Comment (EstherLerouzic) [inline ietf-layer0-types-ext.yang:769]: shouldn't this be a decimal value ?
unit is missing

File: ietf-layer0-types-ext.yang (hunk 1 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:698]: To be discussed with other co-authors in the upcoming weekly call (after the I-D submission)

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: ietf-layer0-types-ext.yang (hunk 3 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:769]: To be discussed with other co-authors in the upcoming weekly call (after the I-D submission): see issue #39 

Co-Authored-By: sergio belotti [sergio.belotti@nokia.com](mailto:sergio.belotti@nokia.com)

File: ietf-layer0-types-ext.yang (hunk 2 above)
Comment (italobusi) [reply ietf-layer0-types-ext.yang:731]: Addressed in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/36/commits/ff8a58466944a4ad32b0c506256f83c5713ee7de

**Pull request merged**